observer/probers/crl/crl_conf_test.go

```diff
@@ -0,0 +1,146 @@
+package probers
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/letsencrypt/boulder/observer/probers"
+	"github.com/prometheus/client_golang/prometheus"
+	"gopkg.in/yaml.v3"
+)
+
+func TestCRLConf_MakeProber(t *testing.T) {
+	conf := CRLConf{}
+	colls := conf.Instrument()
+	badColl := prometheus.Collector(prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "obs_crl_foo",
+			Help: "Hmmm, this shouldn't be here...",
+		},
+		[]string{},
+	))
+	type fields struct {
+		URL string
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		colls   map[string]prometheus.Collector
+		wantErr bool
+	}{
+		// valid
+		{"valid fqdn", fields{"http://example.com"}, colls, false},
+		{"valid fqdn with path", fields{"http://example.com/foo/bar"}, colls, false},
+		{"valid hostname", fields{"http://example"}, colls, false},
+		// invalid
+		{"bad fqdn", fields{":::::"}, colls, true},
+		{"missing scheme", fields{"example.com"}, colls, true},
+		{
+			"unexpected collector",
+			fields{"http://example.com"},
+			map[string]prometheus.Collector{"obs_crl_foo": badColl},
+			true,
+		},
+		{
+			"missing collectors",
+			fields{"http://example.com"},
+			map[string]prometheus.Collector{},
+			true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			c := CRLConf{
+				URL: tt.fields.URL,
+			}
+			if _, err := c.MakeProber(tt.colls); (err != nil) != tt.wantErr {
+				t.Errorf("CRLConf.Validate() error = %v, wantErr %v", err, tt.wantErr)
+			}
+		})
+	}
+}
+
+func TestCRLConf_Instrument(t *testing.T) {
+	t.Run("instrument", func(t *testing.T) {
+		conf := CRLConf{}
+		colls := conf.Instrument()
+		if colls == nil {
+			t.Errorf("crl prober defines metrics but received nil collector map")
+			return
+		}
+		var nu, tu, rcc *prometheus.GaugeVec
+		for name, coll := range colls {
+			switch name {
+			case nextUpdateName:
+				_, ok := coll.(*prometheus.GaugeVec)
+				if !ok {
+					t.Errorf("CRLConf.Instrument() returned collector '%s' of wrong type, got: %T, expected *prometheus.GaugeVec", name, nu)
+					return
+				}
+				nu = coll.(*prometheus.GaugeVec)
+			case thisUpdateName:
+				_, ok := coll.(*prometheus.GaugeVec)
+				if !ok {
+					t.Errorf("CRLConf.Instrument() returned collector '%s' of wrong type, got: %T, expected *prometheus.GaugeVec", name, tu)
+					return
+				}
+				tu = coll.(*prometheus.GaugeVec)
+			case certCountName:
+				_, ok := coll.(*prometheus.GaugeVec)
+				if !ok {
+					t.Errorf("CRLConf.Instrument() returned collector '%s' of wrong type, got: %T, expected *prometheus.GaugeVec", name, rcc)
+					return
+				}
+				rcc = coll.(*prometheus.GaugeVec)
+			default:
+				t.Errorf("CRLConf.Instrument() returned unexpected collector '%s'", name)
+				return
+			}
+		}
```

Comment:
All of this logic duplicates the assertions in MakeProber(). I think you might be better off writing a test that collects the output of Instrument() and then passing it to MakeProber(). You could ensure that there are no errors and also do a nil check on the various metric fields of the resulting CRLProber.

Reply:
Done! Since there's already a test for MakeProber() that needs to call Instrument() to create its test cases, I just added the nil checks into that, to the same effect. Though if there's a different way you'd prefer that to be written, lmk